File: app/src/main/java/com/trevorwiebe/trackacow/db/dao/ArchivedLotDao.java
package com.trevorwiebe.trackacow.db.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.trevorwiebe.trackacow.db.entities.ArchivedLotEntity;

import java.util.List;

@Dao
public interface ArchivedLotDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertArchivedLotEntity(ArchivedLotEntity archivedLotEntity);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertArchivedLotEntityList(List<ArchivedLotEntity> archivedLotEntities);

    @Query("SELECT * FROM archivedLot")
    List<ArchivedLotEntity> getArchiveLots();

    @Query("SELECT * FROM archivedLot WHERE lotId = :lotId")
    ArchivedLotEntity getArchivedLotById(String lotId);

    @Update
    void updateArchivedLot(ArchivedLotEntity archivedLotEntity);

    @Delete
    void deleteArchivedLot(ArchivedLotEntity archivedLotEntity);

    @Query("DELETE FROM archivedLot")
    void deleteArchivedLotTable();

}

File: app/src/main/java/com/trevorwiebe/trackacow/data/local/dao/ArchivedLotDao.kt
package com.trevorwiebe.trackacow.data.local.dao

import androidx.room.*
import com.trevorwiebe.trackacow.data.entities.ArchivedLotEntity

@Dao
interface ArchivedLotDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertArchiveLot(archivedLotEntity: ArchivedLotEntity): Long

    // Deprecated Functions
    @Deprecated("use suspend function")
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertArchivedLotEntity(archivedLotEntity: ArchivedLotEntity?)

    @Deprecated("use suspend function")
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertArchivedLotEntityList(archivedLotEntities: List<ArchivedLotEntity?>?)

    @Deprecated("use flow function")
    @Query("SELECT * FROM archivedLot")
    fun getArchiveLots(): List<ArchivedLotEntity?>?

    @Deprecated("use flow function")
    @Query("SELECT * FROM archivedLot WHERE lotId = :lotId")
    fun getArchivedLotById(lotId: String?): ArchivedLotEntity?

    @Deprecated("use suspend function")
    @Update
    fun updateArchivedLot(archivedLotEntity: ArchivedLotEntity?)

    @Deprecated("use suspend function")
    @Delete
    fun deleteArchivedLot(archivedLotEntity: ArchivedLotEntity?)

    @Deprecated("use suspend function")
    @Query("DELETE FROM archivedLot")
    fun deleteArchivedLotTable()
}